lib/Repository.js

```diff
@@ -172,6 +172,19 @@ class Repository extends Requestable {
       return this._request('GET', `/repos/${this.__fullname}/commits`, options, cb);
    }
 
+    /**
+     * Gets a single commit information for a repository
+     * @see https://developer.github.com/v3/repos/commits/#get-a-single-commit
+     * @param {string} ref - the reference for the commit, if not provided this method behaves as listCommits
+     * @param {string} ref - the reference for the commit
+     * @param {Requestable.callback} cb - will receive the commit information
+     * @return {Promise} - the promise for the http request
+     */
+    getSingleCommit(ref, cb) {
+       ref = ref || '';
```

Comment:
I would prefer that we fail here if the ref isn't provided to avoid callers getting unexpected returns (`getSingleCommit` sounds like I'll be expecting to receive an object, not an array)